lib/docs/initial_schema_proposal.md

```diff
@@ -0,0 +1,260 @@
+# Schema Proposal for Persisting Statistics
```

Comment:
Anyone that is tracking - Can I pls get your review/approval on this 'Initial Schema Proposal'.

cc: @RobAWilkinson @kfrz

Reply:
Move this to a top level `docs/` directory